### PR TITLE
Add provider versions constraint for GCP and Aliyun and support values file customization for tidb-operator and tidb-cluster

### PR DESCRIPTION
**Pull request by DanielZhangQD**
### What problem does this PR solve? 
fix https://github.com/pingcap/tidb-operator/issues/958
### What is changed and how does it work?
Add version constraint for providers.
Modify terraform scripts to support values customization for tidb-operator and tidb-cluster.
### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)
    - `terraform init` does not prompt for version constraint any more
    - tidb-operator can be customized with specific values.yaml
    - tidb-cluster can be customized with specific values.yaml

Code changes

 - Has terraform scripts change
 - Has documents change

Side effects

 NONE

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Does this PR introduce a user-facing change?:
 
 ```release-note
tidb-operator and tidb-cluster can be deployed with customized values.yaml correctly.
 ```

### CODE REVIEW

File: deploy/gcp/main.tf [hunk 1]
```diff
@@ -50,7 +50,7 @@ module "tidb-operator" {
   kubeconfig_path               = local.kubeconfig
   tidb_operator_version         = var.tidb_operator_version
   maintenance_window_start_time = var.maintenance_window_start_time
-  operator_helm_values          = var.operator_helm_values
+  operator_helm_values          = var.operator_helm_values == "" ? "" : file(var.operator_helm_values)
```
Comment (cofyc) [inline deploy/gcp/main.tf]: inline yaml values is sometimes convenient, short and no external file required. How about adding a new variable for file support?

```
operator_helm_values = <<EOF
operatorImage: quay.io/cofyc/tidb-operator:canary
imagePullPolicy: Always
EOF
```

Comment (DanielZhangQD) [reply deploy/gcp/main.tf]: OK, I thought it was designed for external values file, will revert the change.

**Comment by DanielZhangQD** [comment]
@cofyc @aylei PTAL again, revert the change for values customization and just keep the version constraint.

**Comment by DanielZhangQD** [comment]
@cofyc @aylei @jlerche PTAL again, add new variables for customized values file for tidb-operator and tidb-cluster.

**Comment by DanielZhangQD** [comment]
@aylei CI passed, PTAL again.

**Pull request merged**